File: kotlin-spring/src/main/kotlin/traveltimeplatform/models/RequestTimeFilter.kt
package traveltimeplatform.models

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import traveltimeplatform.models.RequestLocation
import traveltimeplatform.models.RequestTimeFilterArrivalSearch
import traveltimeplatform.models.RequestTimeFilterDepartureSearch
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size

/**
 * 
 * @param locations 
 * @param departureSearches 
 * @param arrivalSearches 
 */
data class RequestTimeFilter (

        @get:NotNull @get:Size(min=1) 
        @JsonProperty("locations") val locations: List<RequestLocation>,
@get:Size(max=10) 
        @JsonProperty("departure_searches") val departureSearches: List<RequestTimeFilterDepartureSearch>? = null,
@get:Size(max=10) 
        @JsonProperty("arrival_searches") val arrivalSearches: List<RequestTimeFilterArrivalSearch>? = null
) {

}

File: fsharp-giraffe-server/traveltimeplatform/src/model/RequestTimeFilter.fs
namespace traveltimeplatform.Model

open System
open System.Collections.Generic
open traveltimeplatform.Model.RequestLocation
open traveltimeplatform.Model.RequestTimeFilterArrivalSearch
open traveltimeplatform.Model.RequestTimeFilterDepartureSearch

module RequestTimeFilter = 

  //#region RequestTimeFilter


  type RequestTimeFilter = {
    Locations : RequestLocation[];
    DepartureSearches : RequestTimeFilterDepartureSearch[];
    ArrivalSearches : RequestTimeFilterArrivalSearch[];
  }
  //#endregion
  